avocado/utils/ssh.py

```diff
@@ -199,13 +239,31 @@ def cmd(self, command, ignore_status=True):
                                ignore_status=ignore_status)
         except process.CmdError as exc:
             if exc.result.exit_status == 255:
-                exc.additional_text = 'SSH connection failed'
+                raise SessionError(self)
             else:
                 exc.additional_text = "Command '%s' failed" % command
                 exc.stderr = exc.result.stderr
                 exc.stdout = exc.result.stdout
             raise exc
 
+    def read_file(self, filename):
+        """
+        Reads a file over the SSH session.
+
+        :param filename: path to the file.
+        :type filename: str
+        :return: the file content.
+        :rtype: str
+        :raise: :class:`FileError` if failed to read the file (e.g file not
+                found or permission denied) or :class:`SessionError` for an
+                error in the ssh session.
+        """
+        try:
+            return self.cmd('cat %s' % filename,
+                            ignore_status=False).stdout_text
+        except process.CmdError as exc:
+            raise FileError(filename, self, exc)
+
```

Comment:
I will be honest in that I find this unnecessary. If we start adding a method for every possible file operation or thing we would want to do on top of a session this module will turn into `aexpect`. Even more: the pull request https://github.com/avocado-framework/aexpect/pull/54 there introduces similar method for `cat` performed on file but at least it is part of aexpect which is the module to use if you need such extended session capabilities.

Reply:
IMO, it is useful and we should consider including this.

But I agree with @pevogam when he says that we should avoid making this for every file operation. `aexpect` PR is a very odd example with 8 methods only for `grep`.

But in this case, I think that having handy methods to read/write files over ssh, can be useful.

Said that I made just two small comments here, but since that, I don't have a big picture this, I'm not going to reject/accept. I will wait for more comments here.

Reply:
@pevogam, @beraldoleal , @clebergnu 

Let me give you context: In working on some tests to QEMU and I plan to contribute whatever improvements to Avocado to make the experience of writing tests better. I didn't even know we had this `aexpect` project, so I am using `avocado.utils.ssh` (i.e. sending improviment to this module).

I don't see any problem of switching to `aexpect`  but I would like to know this is desired direction. A glance at `aexpect` sources made me think if it is really in active development (Why? Poor documentation...).  If yes, should its improvements be prioritized over `avocado.utils.ssh`? Or we are talking about different stuffs?

Reply:
@wainersm I am really glad to hear your context and I think you actually raise more important questions than the ones within the scope of this PR. I wasn't around too much when the decision was made to move a lot of code from `autotest` to `avocado` and in particular the decision to turn `aexpect` into its own separately maintained project so I cannot say much about this (even though it seems to make sense from what I have seen so far). However, now that the code is actually there, my main concern here is not about what code is placed where as much as for reinventing the wheel which seems to happen exactly as I predicted in a previous concern I addressed to @clebergnu and he then further mentioned to @beraldoleal at a similar PR https://github.com/avocado-framework/avocado/pull/3633#issuecomment-595805527.

The only reason behind having an SSH utility here was to offer minimal functionality without the extra dependency. As long as all of us find a solution that doesn't involve rewriting the same code and utilities over and over again, I am fine with it. I was advising for increasing the number of maintainers in `aexpect` before, especially if there are some volunteers, but this is just an idea that crossed my mind. Maybe one could think of integrating aexpect back in avocado (although possibly an overkill) or something else. I still think it should be more beneficial for your time if you don't rewrite this code and simply make calls and help maintain aexpect though, contributing truly new functionality there instead.